crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -189,6 +214,25 @@ impl Triangle2d {
         }
     }
 
+    /// Get the area of the triangle
+    pub fn area(&self) -> f32 {
+        let [a, b, c] = self.vertices;
+        (a.x * (b.y - c.y) + b.x * (c.y - a.y) + c.x * (a.y - b.y)).abs() / 2.0
+    }
+
+    /// Get the perimeter of the triangle
+    pub fn perimeter(&self) -> f32 {
+        let [a, b, c] = self.vertices;
+
+        let ab = a.distance(b);
+        let bc = b.distance(c);
+        let ca = c.distance(a);
+
+        debug_assert!(self.area() != 0.0, "triangle has no area");
```

Comment:
Is this assertion needed in `perimeter`?

Reply:
Removed it for now. It was there because I'm not sure if you want to be able to get the perimeter for degenerate triangles since it's not an actual enclosed space, but I suppose you could also expect it to always return the sum of the sides regardless of the validity of the shape